packages/flutter_tools/templates/app/pubspec.yaml.tmpl

```diff
@@ -27,6 +27,11 @@ dependencies:
 
 {{#withPluginHook}}
   {{pluginProjectName}}:
+    # The example app is bundled with the plugin so we use a path dependency on
+    # the parent directory to use the current plugin's version. 
+    # When depending on this package from a real application you should use:
```

Comment:
I would put this part first since it's the most important part of this comment. Also consider doing something to draw the readers eye to it.

Reply:
Moving the `When depending on this package...`-part first makes sense.